pilot/pkg/serviceregistry/serviceentry/ip_autoallocation.go

```diff
@@ -0,0 +1,104 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package serviceentry
+
+import (
+	"encoding/json"
+	"net/netip"
+
+	"istio.io/api/meta/v1alpha1"
+	networkingv1alpha3 "istio.io/client-go/pkg/apis/networking/v1alpha3"
+	"istio.io/istio/pilot/pkg/features"
+	"istio.io/istio/pkg/config/constants"
+)
+
+const (
+	IPAutoallocateStatusType = "ip-autoallocate"
+)
+
+func GetV2AddressesFromServiceEntry(se *networkingv1alpha3.ServiceEntry) []netip.Addr {
+	if se == nil {
+		return []netip.Addr{}
+	}
+	return kludgeFromStatus(se.Status.GetConditions())
+}
+
+func ShouldV2AutoAllocateIP(se *networkingv1alpha3.ServiceEntry) bool {
```

Comment:
Do we also need to skip for NONE mode and/or wildcards? If that is handled elsewhere, can we add a comment indicating?